### ISSUE TITLE
Bug: false positive "@types/jest" using jest and typescript

### ISSUE THREAD

**Issue by Barkole** [depcheck/depcheck#553]
Jest requires "@types/jest" to have typescript support

**Comment by VladimirGrenaderov** [comment]
It's a common issue with @types/xxx, if you go via simple way and allow TS to grab all packages from @types scope - you never can answer the question 'which types I use now? what if it works just because side-effect?". Automatic dependency detection is impossible too (I mean more or less simple way different from full compilation).

Smarter way - to add required type references into tsconfig.json (vscode accepts those changes only after reload, just FYI). I have special for tsconfig that analyzed types field, and hope PR will be appear shortly.

**Comment by slambdev** [comment]
Using the latest version (1.3.1) and I"m still getting false positives with `@types/jest`. All other `@types` modules are correctly detected by depcheck. Any chance for an update on this?

**Comment by rumpl** [comment]
@slambdev any chance you could create a repo or link to one that reproduces this error please?

**Comment by slambdev** [comment]
Unfortunately, our repo is private. Example of how we are calling it is:

```
depcheck . --ignore-bin-package=true --specials=eslint,tslint,jest,ttypescript
```
When I run it with the `--json`, I get the following

```
{
    "dependencies": [],
    "devDependencies": [
        "@types/jest",
        "eslint-import-resolver-typescript"
    ],
    "missing": {},
    "using": {
        "@typescript-eslint/parser": [
            "/Users/shannen/Documents/forks/guest-promotion/.eslintrc.js"
        ],
        "@typescript-eslint/eslint-plugin": [
            "/Users/shannen/Documents/forks/guest-promotion/.eslintrc.js"
        ],
        "eslint-plugin-jest": [
            "/Users/shannen/Documents/forks/guest-promotion/.eslintrc.js"
        ],
        "eslint-plugin-simple-import-sort": [
            "/Users/shannen/Documents/forks/guest-promotion/.eslintrc.js"
        ],
        "eslint-plugin-prettier": [
            "/Users/shannen/Documents/forks/guest-promotion/.eslintrc.js"
        ],
        "eslint-plugin-import": [
            "/Users/shannen/Documents/forks/guest-promotion/.eslintrc.js"
        ],
        "eslint-config-airbnb-base": [
            "/Users/shannen/Documents/forks/guest-promotion/.eslintrc.js"
        ],
        "eslint-config-prettier": [
            "/Users/shannen/Documents/forks/guest-promotion/.eslintrc.js"
        ],
        "newrelic": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "@types/newrelic": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "@slack/bolt": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/routes/emailRoutes.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "@slack/types": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts"
        ],
        "config": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/newrelic.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackAdminClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackSCIMClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/templates/campaignCreationTemplate.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/email/emailClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/campaignNotificationTemplate.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/promotionNotificationTemplate.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/guestPromotion/guestPromoter.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/newrelic.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "@types/config": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/newrelic.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackAdminClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackSCIMClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/templates/campaignCreationTemplate.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/email/emailClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/campaignNotificationTemplate.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/promotionNotificationTemplate.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/guestPromotion/guestPromoter.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/newrelic.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "dateformat": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/email/emailMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/campaignNotificationTemplate.ts"
        ],
        "@types/dateformat": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/email/emailMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/campaignNotificationTemplate.ts"
        ],
        "memoizee": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.ts"
        ],
        "@types/memoizee": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.ts"
        ],
        "mongoose": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/database/dbService.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/database/schemas/campaignSchema.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/database/schemas/userSchema.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "@types/mongoose": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/database/dbService.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/database/schemas/campaignSchema.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/database/schemas/userSchema.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "slack-at-ibm-bots-utilities": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackAdminClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackSCIMClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/email/emailClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/routes/emailRouter.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/routes/emailRoutes.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/guestPromotion/guestPromoter.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/guestPromotion/guestPromoter.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "sigsci-module-nodejs": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts"
        ],
        "@slack/web-api": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackAdminClient.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackAdminClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackMessenger.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "ibm-cos-sdk": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/cron/remindGuestsAndCloseCampaign.ts"
        ],
        "body-parser": [
            "/Users/shannen/Documents/forks/guest-promotion/src/bolt-app.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/routes/emailRoutes.spec.ts"
        ],
        "limiter": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackAdminClient.ts"
        ],
        "lodash.partition": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/email/emailMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/guestPromotion/guestPromoter.ts"
        ],
        "@types/lodash.partition": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/email/emailMessenger.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/guestPromotion/guestPromoter.ts"
        ],
        "axios": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackSCIMClient.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackSCIMClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/routes/emailRoutes.spec.ts"
        ],
        "@types/axios": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackSCIMClient.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackSCIMClient.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/routes/emailRoutes.spec.ts"
        ],
        "ts-jest": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackSCIMClient.spec.ts"
        ],
        "@types/node": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.spec.ts",
            "/Users/shannen/Documents/forks/guest-promotion/config/production.ts"
        ],
        "csvtojson": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.ts"
        ],
        "node-gzip": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.ts"
        ],
        "@types/node-gzip": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/slack/slackWorkspaceProvider.ts"
        ],
        "@sendgrid/mail": [
            "/Users/shannen/Documents/forks/guest-promotion/src/utilities/email/emailClient.ts"
        ],
        "markdown-it": [
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/campaignNotificationTemplate.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/promotionNotificationTemplate.ts"
        ],
        "@types/markdown-it": [
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/campaignNotificationTemplate.ts",
            "/Users/shannen/Documents/forks/guest-promotion/src/templates/promotionNotificationTemplate.ts"
        ],
        "express": [
            "/Users/shannen/Documents/forks/guest-promotion/src/routes/emailRoutes.ts"
        ],
        "express-async-handler": [
            "/Users/shannen/Documents/forks/guest-promotion/src/routes/emailRoutes.ts"
        ]
    },
    "invalidFiles": {},
    "invalidDirs": {}
}
```

**Comment by crossjs** [comment]
not stale
